packages/@aws-cdk/aws-codepipeline-actions/lib/s3/deploy-action.ts

```diff
@@ -29,6 +29,24 @@ export interface S3DeployActionProps extends codepipeline.CommonAwsActionProps {
    * The Amazon S3 bucket that is the deploy target.
    */
   readonly bucket: s3.IBucket;
+
+  /**
+   * The specified canned ACL to objects deployed to Amazon S3.
+   * This overwrites any existing ACL that was applied to the object.
+   *
+   * @default - the original object ACL
+   */
+  readonly cannedAcl?: s3.BucketAccessControl;
+
+  /**
+   * The caching behavior for requests/responses for objects in the bucket.
+   *
+   * @see https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.9 Cache-Control header field for HTTP operations
+   * @example 'public, max-age=31536000'
```

Comment:
Hmm. I wonder whether we can provide a bit of sugar for this property? Maybe something like (just spitballing here):

```ts
new cpactions.S3DeployAction({
  actionName: 'S3Deploy',
  bucket,
  sourceOutput,
 cacheControl: s3.CacheControl.public({ maxAge: Duration.hours(2) })`
  ...
```

? We could have a `CacheControl.of(string)` as an "escape mechanism" to be able to provide the string directly in case our API doesn't implement the entire RFC2616.

Thoughts on this idea?

Reply:
I didn't think about it, I think it's great. Let me get to it

Reply:
I'm not sure putting it in the `aws-s3` package is ideal though. It could be used by CloudFront's WebDistribution, but that's the only other one I can see:

https://github.com/aws/aws-cdk/blob/46b32927efbe90d16f0129770c6a0c350855f9e2/packages/%40aws-cdk/aws-cloudfront/lib/web_distribution.ts#L168

Where should I put the `CacheControl` construct?

Reply:
I ended up going with a builder pattern, as you can chain the statements:

```ts
(new CacheControlResponse()).of('immutable').public().sMaxAge(Duration.hours(3));
```

Reply:
Alright, so a couple of things:

* I'm fairly sure there is no point in setting Cache-Control request headers values for S3, as it'll always send a request. I implemented both `CacheControlRequest` and `CacheControlResponse` just in case. We can get rid of the latter if there is no use for them in the CDK
* I just dumped the classes at the end of `s3/deploy-actions.ts`, but we should find a better home for them

Reply:
This is super cool @nmussy , thanks!

@eladb @RomainMuller can you chime in here - what do you think of the proposed API of `CacheControl`, and what module should we put it in?